nushell_completions.go

```diff
@@ -0,0 +1,128 @@
+// limitations under the License.
+
+package cobra
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+)
+
+func (c *Command) GenNushellCompletion(w io.Writer, includeDesc bool) error {
+	buf := new(bytes.Buffer)
+	WriteStringAndCheck(buf, "# nushell completion -*- shell-script -*- \n")
+	WriteStringAndCheck(buf, fmt.Sprintf(`
+let cobra_completer = {|spans|
+    let ShellCompDirectiveError = %[1]d
+    let ShellCompDirectiveNoSpace = %[2]d
+    let ShellCompDirectiveNoFileComp = %[3]d
+    let ShellCompDirectiveFilterFileExt = %[4]d
+    let ShellCompDirectiveFilterDirs = %[5]d
+    let ShellCompDirectiveKeepOrder = %[6]d
+
+    let cmd = $spans | first 
+    let rest = $spans | skip
+
+    def exec_complete [
+        spans: list<string>
+    ] {
+        # This will catch the stderr message related to the directive and any other errors,
+        # such as the command not being a cobra based command
+        let result = do --ignore-errors { COBRA_ACTIVE_HELP=0 run-external $cmd "__complete" ...$spans | complete }
+
+        if $result != null and $result.exit_code == 0 {
+            let completions = $result.stdout | lines
+
+            # the directive is the last line
+            let directive = do -i { $completions | last | str replace ':' '' | into int }
+
+            let completions = $completions | drop | each { |it| 
+                # the first word is the command, the rest is the description
+                let words = $it | split row -r '\s{1}'
+
+                # If the last span contains a hypen and equals, attach it to the name
+                let last_span = $spans | last
+                let words = if ($last_span =~ '^-') and ($last_span =~ '=$') {
+                    $words | each {|it| $"($last_span)($it)" }
+                } else {
+                    $words
+                }
+
+                {value: ($words | first | str trim), description: ($words | skip | str join ' ')}
+            }
+
+            {completions: $completions, directive: $directive}
+        } else {
+            {completions: [], directive: -1}
+        }
+    }
+
+    if (not ($rest | is-empty)) {
+        let result = exec_complete $rest
+        let completions = $result.completions
+        let directive = $result.directive
+
+        # Add space at the end of each completion
+        let completions = if $directive != $ShellCompDirectiveNoSpace {
+            $completions | each {|it| {value: $"($it.value) ", description: $it.description}}
+        } else {
+            $completions
+        }
+
+        # Cobra returns a list of completions that are supported with this directive
+        # There is no way to currently support this in a nushell external completer
+        let completions = if $directive == $ShellCompDirectiveFilterFileExt {
+            []
+        } else {
+            $completions
+        }
+
+        if $directive == $ShellCompDirectiveNoFileComp {
+            # Allow empty results as this will stop file completion
```

Comment:
This does not work for me.  
I'm running nushell 0.100.0

You can try it:

```
# Notice the directive returned is ShellCompDirectiveNoFileComp
> kubectl __complete help annotate ''
:4
Completion ended with directive: ShellCompDirectiveNoFileComp

# But the following does shell completion
> kubectl help annotate <tab>
```

If we can have a `debug` function to print to a file like my above comments, then I can try to debug such things.

Reply:
fixed